test/crossfilter-test.js

```diff
@@ -1815,7 +1815,7 @@ suite.addBatch({
             var group = data.tags.groupAll().reduceCount();
             data.total.filterRange([200, Infinity]);
             data.total.filterFunction(function(d) { return "0"; });
-            assert.equal(group.value(), 43);
+            assert.equal(group.value(), 119);
```

Comment:
I think this is a change in functionality. See 1.4.2 behavior here: https://jsfiddle.net/esjewett/8nhy71d5/1/

However this is inconsistent with the behavior for `dimension.group` under array dimensions. See https://jsfiddle.net/esjewett/vaj08pa0/

I believe that the `dimension.group` case under 1.4.2 is a bug while the `dimension.groupAll` behavior is correct. A single record should only ever be counted once per group. However, I'm open to discussion of this.

Regardless, I feel that this may be a breaking change for some people so we should consider it.

Reply:
Looking back at 1.4.2 it looks like groupAll just counted up the unfiltered records. See [https://jsfiddle.net/2rvr4dbs/](https://jsfiddle.net/2rvr4dbs/)

I'm not sure I see much value in only counting a single record per group when it comes to groupAll and array dimensions. If you take a normal dimension, group it and add up all of the values, it will equal the groupAll value for that dimension. However, if you try and do the same with an array dimension in 1.4.2, that isn't the case.

Either way, it is currently broken in 1.4.2.

Reply:
That's a good point. If you just want to count up all records, you can do `groupAll` on a normal dimension. Similarly, you can control duplicate array values in normal groups in your dimension accessor, so it's probably fine to double-count rows within groups as the user of Crossfilter has control over this. So, ok, I'm alright with this change.

@gordonwoodhull and @tannerlinsley, can you read over this and give an OK if you are interested?

Secondary discussion: Can we do this in a patch or minor release. I suspect no one is using this functionality at the moment, so I'm tempted to do so. I wish we'd caught this in the 1.4 alphas though :(